x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java

```diff
@@ -0,0 +1,941 @@
+     *       "type": "integer",
+     *       "ignore_malformed": true/false
+     *     }
+     *   }
+     * }
+     * </pre>
+     */
+    public void testTextFieldWithIntegerSubfield() throws IOException {
+        int value = randomInt();
+        Map<String, Object> result = textTest().sub("int", intTest()).roundTrip(value);
+
+        assertMap(
+            result,
+            matchesMap().entry("columns", List.of(columnInfo("text_field", "text"), columnInfo("text_field.int", "integer")))
+                .entry("values", List.of(matchesList().item(Integer.toString(value)).item(value)))
+        );
+    }
+
+    /**
+     * <pre>
+     * "text_field": {
+     *   "type": "text",
+     *   "fields": {
+     *     "int": {
+     *       "type": "integer",
+     *       "ignore_malformed": true
+     *     }
+     *   }
+     * }
+     * </pre>
+     */
+    public void testTextFieldWithIntegerSubfieldMalformed() throws IOException {
+        String value = randomAlphaOfLength(5);
+        Map<String, Object> result = textTest().sourceMode(SourceMode.DEFAULT).sub("int", intTest().ignoreMalformed(true)).roundTrip(value);
+
+        assertMap(
+            result,
+            matchesMap().entry("columns", List.of(columnInfo("text_field", "text"), columnInfo("text_field.int", "integer")))
+                .entry("values", List.of(matchesList().item(value).item(null)))
+        );
+    }
+
+    /**
+     * <pre>
+     * "text_field": {
+     *   "type": "text",
+     *   "fields": {
+     *     "ip": {
+     *       "type": "ip",
+     *       "ignore_malformed": true/false
+     *     }
+     *   }
+     * }
+     * </pre>
+     */
+    public void testTextFieldWithIpSubfield() throws IOException {
+        String value = NetworkAddress.format(randomIp(randomBoolean()));
+        Map<String, Object> result = textTest().sub("ip", ipTest()).roundTrip(value);
+
+        assertMap(
+            result,
+            matchesMap().entry("columns", List.of(columnInfo("text_field", "text"), columnInfo("text_field.ip", "ip")))
+                .entry("values", List.of(matchesList().item(value).item(value)))
+        );
+    }
+
+    /**
+     * <pre>
+     * "text_field": {
+     *   "type": "text",
+     *   "fields": {
+     *     "ip": {
+     *       "type": "ip",
+     *       "ignore_malformed": true
+     *     }
+     *   }
+     * }
+     * </pre>
+     */
+    public void testTextFieldWithIpSubfieldMalformed() throws IOException {
+        String value = randomAlphaOfLength(10);
+        Map<String, Object> result = textTest().sourceMode(SourceMode.DEFAULT).sub("ip", ipTest().ignoreMalformed(true)).roundTrip(value);
+
+        assertMap(
+            result,
+            matchesMap().entry("columns", List.of(columnInfo("text_field", "text"), columnInfo("text_field.ip", "ip")))
+                .entry("values", List.of(matchesList().item(value).item(null)))
+        );
+    }
+
+    /**
+     * <pre>
+     * "int_field": {
```

Comment:
I'll clean this one up. I always write `int` instead of `integer` when I work with that mapper. Not sure why. OTOH, I tend to use `str` when I mean `text|keyword`. Old habit. I'll double check it.